src/components/spatial/SpatialSubscriber.js

```diff
@@ -116,27 +126,41 @@ export default function SpatialSubscriber(props) {
   const [raster, imageLayerLoaders] = useRasterData(
     loaders, dataset, setItemIsReady, addUrl, false,
     autoImageLayers => setAutoLayers(prev => (
-      { [dataset]: [...(prev[dataset] || []), ...autoImageLayers] }
+      // This prevents old updates from overriding the current dataset.
+      // The previous state must be for this dataset, otherwise it's an old update.
+      Object.keys(prev).includes(dataset)
+        ? { [dataset]: [...(prev[dataset] || []), ...autoImageLayers] }
+        : prev
```

Comment:
As noted in the comment, due to the asynchronicity of it all, we were running into a situation where the previous `dataset`'s raster `autoImageLayers` (if you change to quickly) was being used here and then completely overriding the current auto-configuration (which includes the above `cells` etc.). This was problematic since then the next auto-layer update (i.e for the current `dataset`) would override that leaving you with no `cells` from above, for example.